{
  try {
    return future.get(time,unit);
  }
 catch (  InterruptedException|TimeoutException|ExecutionException e) {
    LOG.log(Level.SEVERE,"Exception processing future ",e);
    future.cancel(true);
    return null;
  }
}
